### ISSUE TITLE
Empty files issue related with url redirects

### ISSUE THREAD

**Issue by neddihrehat** [andresdavid90/pizza-guy#19]
Just downloaded a sample gist from github. And it returned empty files. 

The code:
```js
var pizzaGuy = require('pizza-guy');
const files = [
  'https://github.com/andresdavid90/pizza-guy/archive/master.zip'
];
pizzaGuy
  .deliver(files)
  .onAddress('./downloads')
  .onSuccess((info) => {
    console.log(`${info.fileName} succeed!`);
  })
  .onError((info) => {
    console.log(`${info.fileName} failed`);
  })
  .start();
```

The result is a empty file.
http://i.imgur.com/aalTtaw.png

Probably issue with **https** and/or **redirects**?

**Comment by andresdavid90** [comment]
@neddihrehat Thank you for reporting this. I was already able to replicate the issue and I'm currently investigating.

**Comment by andresdavid90** [comment]
We injected the issue in the commit [beb6883458a3e785d186b54c0290679c7b5156c4](https://github.com/andresdavid90/pizza-guy/commit/beb6883458a3e785d186b54c0290679c7b5156c4).
Still investigating.

**Comment by andresdavid90** [comment]
Well, actually the commit that I just mentioned injected an issue that is affecting any kind of file, but even if we fix the issue, zip files with redirections like `https://github.com/andresdavid90/pizza-guy/archive/master.zip` are still failing.

I'm creating a separate issue for the big bug affecting everything, and we'll use this one to track files with **url redirects**.

**Comment by andresdavid90** [comment]
@neddihrehat I just fixed this issue in #37. We'll have this fix soon in master so it would be reflected in NPM soon. Thanks for reporting this. 🤗
